probe-rs-tools/src/bin/probe-rs/cmd/dap_server/debug_adapter/dap/request_helpers.rs

```diff
@@ -9,14 +9,96 @@ use capstone::{
     arch::arm::ArchMode as armArchMode, arch::arm64::ArchMode as aarch64ArchMode,
     arch::riscv::ArchMode as riscvArchMode, prelude::*, Endian,
 };
+use once_cell::sync::Lazy;
 use probe_rs::{
     debug::{ColumnType, ObjectRef, SourceLocation},
     CoreType, InstructionSet, MemoryInterface,
 };
 use std::{fmt::Write, time::Duration};
+use typed_path::TypedPathBuf;
 
 use super::dap_types::{Breakpoint, InstructionBreakpoint, MemoryAddress};
 
+// Source file mapping for rustlib, e.g. Some(("/rustc/<hash>", "<sysroot>/lib/rustlib/src/rust"))
+// This can be None if rustc is not found or gives bad output
+static RUSTLIB_SOURCE_MAP: Lazy<Option<(TypedPathBuf, TypedPathBuf)>> = Lazy::new(|| {
+    let rustc = rustc_binary();
+
+    // Call rustc --version --verbose to get hash
+    let cmd = subprocess::Exec::cmd(&rustc)
```

Comment:
Could we avoid introducing subprocess just for this?

Reply:
The reason I used subprocess was to add execution time limits since we can't be 100% sure what we're actually running (that the binary called `rustc` is actually rustc).

If you think I'm being unnecessarily cautious here I can just use `std::process::Command` instead?

Reply:
Oh... I'm not quite sure, it wouldn't have occurred to me. I'd probably not worry about it, but who knows.